lib/client-api/src/client_api.ts

```diff
@@ -116,29 +138,24 @@ export default class ClientApi {
   getSeparators = () => ({
     hierarchyRootSeparator: '|',
     hierarchySeparator: /\/|\./,
-    ...this._globalParameters.options,
+    ..._globalParameters.options,
   });
 
-  addDecorator = (decorator: DecoratorFunction) => {
-    this._globalDecorators.push(decorator);
+  addDecorator = (decorator: DecoratorFunction, framework?: string) => {
+    addDecorator(decorator, framework);
   };
 
-  addParameters = (parameters: Parameters | { globalParameter: 'string' }) => {
-    this._globalParameters = {
-      ...this._globalParameters,
-      ...parameters,
-      options: {
-        ...merge(get(this._globalParameters, 'options', {}), get(parameters, 'options', {})),
-      },
-      // FIXME: https://github.com/storybookjs/storybook/issues/7872
-      docs: {
-        ...merge(get(this._globalParameters, 'docs', {}), get(parameters, 'docs', {})),
-      },
-    };
+  addParameters = (parameters: Parameters) => {
+    addParameters(parameters);
   };
 
   clearDecorators = () => {
```

Comment:
Should this function be top-level too?

Reply:
From what I can tell this is more of an internal API that's used by ConfigAPI, and I've never seen any user code use it. We can add it easily if anybody asks for it, and if nobody does, then we probably shouldn't expose it anyway.

Reply:
If users would use this, it would behave very very unpredictable, do not expose this to users